### PR TITLE
[MTDB-11996] Add scratch & win boost on polygon

### PR DESCRIPTION
**Pull request by micahpatng**
- add scratch & win boost button and modal 
- add `bonus` on `Player`

Screenshot 2023-11-29 at 10 44 03
Screenshot 2023-11-29 at 10 44 13

### CODE REVIEW

File: src/components/Boosts/Boosts.tsx [hunk 1]
```diff
@@ -72,34 +90,38 @@ const Boosts: FC<Props> = ({ mobileVersion }) => {
...
-                  <Clock size={16} />
-                ) : (
-                  <>{boost.unlocked ? <Check /> : <Lock />}</>
-                )}
-                {boost.label}
-              </Label>
-              <Boost $unlocked={boost.unlocked}>{boost.desciption}</Boost>
-            </BoostButton>
-          ))}
+          {boostList(player, isPolygon).map(
```
Comment (PorlyBe) [inline src/components/Boosts/Boosts.tsx:103]: consider how we'd handle adding other chains in the future... maybe worth doing `boostList(player, chainId)` and having that return only the bonuses available by a specific chain... can test with different bonuses for 1, 5 and 137

Comment (micahpatng) [reply src/components/Boosts/Boosts.tsx]: that makes sense, fixed!

File: src/components/Boosts/Boosts.tsx [hunk 2]
```diff
@@ -20,27 +20,44 @@ import {
...
+      id: "burn",
+      unlocked: true,
+      show: !isPolygon,
+      label: "Burn",
+      description: "Skip time",
+    },
+    {
+      id: "scratcher",
+      unlocked: true,
+      show: isPolygon,
```
Comment (PorlyBe) [inline src/components/Boosts/Boosts.tsx:55]: maybe just return different arrays depending on chainid... doing this way would get complicated if different chains end up having different copy or bonus values

File: src/components/Boosts/Modals/Scratcher.tsx [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+        <br />
+        2. Scratch the same number 3 times to win VERSE
+        <br />
+        3. Claim your prize
+      </Description>
+      <Text>
+        Your Verse Clicker one-time bonus = your current points production rate
+        X your Scratch & Win prize amount
+      </Text>
+      <LinkButton href="https://scratcher.verse.bitcoin.com/" target="_blank">
```
Comment (PorlyBe) [inline src/components/Boosts/Modals/Scratcher.tsx:24]: will need the isWallet check in here to navigate directly

File: src/context/reducers/player.ts [hunk 4]
```diff
@@ -12,6 +12,7 @@ export type Player = {
   verseHolder: boolean;
   isFarming: boolean;
   isStaking: boolean;
+  bonus: number;
```
Comment (PorlyBe) [inline src/context/reducers/player.ts:15]: player object with bonuses is little different to this... please remove for now, i updated types in my burn branch

**Review by PorlyBe — CHANGES_REQUESTED** [review]

**Review by PorlyBe — APPROVED** [review]

**Pull request merged**